python/pip_install/pip_repository.bzl

```diff
@@ -367,6 +392,40 @@ whl_library = repository_rule(
     doc = """
 Download and extracts a single wheel based into a bazel repo based on the requirement string passed in.
 Instantiated from pip_repository and inherits config options from there.""",
-    implementation = _impl_whl_library,
+    implementation = _whl_library_impl,
     environ = common_env,
 )
+
+def package_annotation(
+        build_content = None,
+        copy_files = {},
+        copy_executables = {},
+        data = [],
+        data_exclude_glob = [],
+        srcs_exclude_glob = []):
+    """Annotations to apply to the BUILD file content from package generated from a `pip_repository` rule.
+
+    [cf]: https://github.com/bazelbuild/bazel-skylib/blob/main/docs/copy_file_doc.md
+
+    Args:
+        build_content (str, optional): Raw text to add to the generated `BUILD` file of a package.
```

Comment:
nit: might be nice to mirror the same attribute names as some builtin repo rules for `build_content` -> `build_file_content` https://docs.bazel.build/versions/main/repo/http.html#http_archive-build_file_content

Reply:
This isn't intended to be the whole build file. Just a small snippet to add to the bottom of the generated one. It's been renamed to `additive_build_content`.